packages/react-core/src/next/components/DualListSelector/DualListSelectorListWrapper.tsx

```diff
@@ -0,0 +1,129 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/DualListSelector/dual-list-selector';
+import { css } from '@patternfly/react-styles';
+import { getUniqueId, handleArrows } from '../../../helpers';
+import { DualListSelectorList } from './DualListSelectorList';
+import { DualListSelectorContext, DualListSelectorListContext } from './DualListSelectorContext';
+
+export interface DualListSelectorListWrapperProps extends React.HTMLProps<HTMLDivElement> {
+  /** Additional classes applied to the dual list selector. */
+  className?: string;
+  /** Anything that can be rendered inside of the list */
+  children?: React.ReactNode;
+  /** ID of the dual list selector list */
+  id?: string;
+  /** Accessibly label for the list */
+  'aria-labelledby': string;
+  /** @hide forwarded ref */
+  innerRef?: React.RefObject<HTMLDivElement>;
+  /** @hide Options to list in the pane. */
+  options?: React.ReactNode[];
+  /** @hide Options currently selected in the pane. */
+  selectedOptions?: string[] | number[];
+  /** @hide Callback for when an option is selected. Optionally used only when options prop is provided. */
+  onOptionSelect?: (e: React.MouseEvent | React.ChangeEvent | React.KeyboardEvent, index: number, id: string) => void;
+  /** @hide Function to determine if an option should be displayed depending on a dynamically built filter value */
+  displayOption?: (option: React.ReactNode) => boolean;
+  /** Flag indicating whether the component is disabled. */
+  isDisabled?: boolean;
+}
+
+export const DualListSelectorListWrapperBase: React.FunctionComponent<DualListSelectorListWrapperProps> = ({
+  className,
+  children,
+  'aria-labelledby': ariaLabelledBy,
+  innerRef,
+  options = [],
+  selectedOptions = [],
+  onOptionSelect,
+  displayOption,
+  id = getUniqueId('dual-list-selector-list'),
+  isDisabled = false,
+  ...props
+}: DualListSelectorListWrapperProps) => {
+  const [focusedOption, setFocusedOption] = React.useState('');
+  const ref = React.useRef(null);
+  const menuRef = innerRef || ref;
+  const { isTree } = React.useContext(DualListSelectorContext);
+
+  // sets up keyboard focus handling for the dual list selector menu child of the pane. This keyboard
+  // handling is applied whether the pane is dynamically built or passed via the children prop.
+  const handleKeys = (event: KeyboardEvent) => {
```

Comment:
Is this function ever used anymore?

Reply:
yeah this is used for handling key actions instead of mouse in the dual list selector panes - i removed the confusing part of the comment to make it clearer